qiskit/providers/fake_provider/fake_backend.py

```diff
@@ -341,8 +342,6 @@ def run(self, run_input, **options):
                 "Invalid input object %s, must be either a "
                 "QuantumCircuit, Schedule, or a list of either" % circuits
             )
-        if pulse_job:  # pulse job
-            raise QiskitError("Pulse simulation is currently not supported for V2 fake backends.")
```

Comment:
You need to keep these lines. To run pulse job, you need to do the same things as in [FakeBackend(BackendV1).run](https://github.com/Qiskit/qiskit-terra/blob/16c2d31df096e154c5ebf708dbbefc9e82233c95/qiskit/providers/fake_provider/fake_backend.py#L548). However, PulseSystemModel has not yet support BackendV2 so you cannot to have it. I suggest to change the scope of this PR and focus on just replacing the `_get_noise_model_from_backend_v2` with `from_backend`.